contracts/tokenomics/maker/src/contract.rs

```diff
@@ -185,40 +187,42 @@ pub fn execute(
 }
 
 /// # Description
-/// Collects astro tokens. Before that collects all assets from the all specified
-/// pairs and performs a swap operation for all non-astro tokens into an astro token.
+/// Collects astro tokens. Before that collects all assets and performs a swap operation for all
+/// non-astro tokens into an astro token.
 /// Returns an [`ContractError`] on failure, otherwise returns the [`Response`] object if the
 /// operation was successful.
 /// # Params
 /// * **deps** is the object of type [`DepsMut`].
 ///
 /// * **env** is the object of type [`Env`].
 ///
-/// * **pair_addresses** is a vector that contains object of type [`Addr`].
-/// Sets the pairs for which the collect operation will be performed.
-fn collect(deps: DepsMut, env: Env, pair_addresses: Vec<Addr>) -> Result<Response, ContractError> {
+/// * **assets** is a vector that contains object of type [`AssetWithLimit`].
+/// Sets the assets for which the collect operation will be performed.
+fn collect(
+    deps: DepsMut,
+    env: Env,
+    assets: Vec<AssetWithLimit>,
+) -> Result<Response, ContractError> {
     let cfg = CONFIG.load(deps.storage)?;
 
     let astro = token_asset_info(cfg.astro_token_contract.clone());
 
-    // Collect assets
-    let mut assets_map: HashMap<String, AssetInfo> = HashMap::new();
-    for pair in pair_addresses {
-        let pair = query_pair(deps.as_ref(), pair)?;
-        assets_map.insert(pair[0].to_string(), pair[0].clone());
-        assets_map.insert(pair[1].to_string(), pair[1].clone());
```

Comment:
You can check for duplicates a bit easier. Just collect Vec into HashSet  and compare their length. If they are not equal then there are duplicates in Vec.